pkg/controller/controller_instance.go

```diff
@@ -40,12 +40,29 @@ func (c *controller) instanceAdd(obj interface{}) {
 		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
-	// TODO(vaikas): If the obj (which really is an ServiceInstance right?) has
-	// AsyncOpInProgress flag set, just add it directly to c.pollingQueue
-	// here? Why shouldn't we??
+
 	c.instanceQueue.Add(key)
 }
 
+func (c *controller) instanceUpdate(oldObj, newObj interface{}) {
```

Comment:
I don't recall if this is called only with `*v1alpha1.ServiceInstance`s. If so, can you change the param types to that? Otherwise, can you change the parameter types to be `runtime.Object`?

Reply:
I don't think we can change that; it's part of how the informer event handlers work. The functions are passed to the informers as type `ResourceEventHandlerFuncs`, which is defined in `vendor/k8s.io/client-go/tools/cache/controller.go` as the following struct:

```
type ResourceEventHandlerFuncs struct {
        AddFunc    func(obj interface{})
        UpdateFunc func(oldObj, newObj interface{})
        DeleteFunc func(obj interface{})
}
```